src/server.c

```diff
@@ -5426,6 +5426,23 @@ int processCommand(client *c) {
         return C_OK;
     }
 
+    /* During async-loading, block certain commands.
+     * TODO: this should be promoted to a command flag, same as lua_timedout below. */
+    if (server.async_loading && (
+        c->cmd->proc == saveCommand ||
+        c->cmd->proc == bgsaveCommand ||
+        c->cmd->proc == saveCommand ||
+        c->cmd->proc == bgrewriteaofCommand ||
+        c->cmd->proc == clusterCommand || /* I'm not sure that's necessary */
+        c->cmd->proc == sentinelCommand || /* I'm not sure that's necessary */
+        c->cmd->proc == moduleCommand ||
+        c->cmd->proc == syncCommand ||
+        c->cmd->proc == replicaofCommand))
```

Comment:
I'd like to understand the specific risk in running each of these commands during async loading.

Reply:
ok, just fix the double `saveCommand`.

Reply:
@hwware thanks, I now realize my question was silly, sentinel was already allowed in loading, so async loading on not doesn't matter. Sorry.

Reply:
From a protocol perspective, SLOTS is required. I know `NODES` and `INFO` are also called by clients, so it would be best to keep those as well. `MYID` is also likely safe to expose, everything else should probably be hidden.

Reply:
thanks. just to make it clear: `s/keep/allow/g`. i.e. they where all blocked till now (together with all other traffic), and now the ones you mentioned will be allowed, and the others will keep being blocked.

Reply:
Also `CLUSTER LINKS` from https://github.com/redis/redis/pull/9774 would be okay to expose during loading.

Reply:
@madolson looking though the rest of the cluster sub-commands, i think these should not be blocked either:
* COUNTKEYSINSLOT
* GETKEYSINSLOT
* KEYSLOT
* HELP

I updated the code and top comment, please ACK.

Reply:
Thinking about this more, I think we should also allow:

*FORGET: A common usecase is you have to call forget on the entire cluster for a node to remove, having one shard fail because it's syncing will just cause the rest of the cluster to learn about the node again. A node can't forget itself, primaries, or replicas, so it should be safe.
*replicas: Get replicas of a node.
*count-failure-reports: Informational command about number of failures.